android/src/main/java/com/matejdr/admanager/BannerAdView.java

```diff
@@ -118,7 +133,114 @@ public void onAdClosed() {
             }
 
         });
+
+        updateLayout();
+
         this.addView(this.adView);
+
+        updateLayout();
+    }
+
+    private class MeasureAndLayoutRunnable implements Runnable {
+        @Override
+        public void run() {
+            updateLayout();
+        }
+    }
+
+    private boolean isFluid() {
+        if (this.adSize == null) {
+            return false;
+        }
+
+        return this.adSize.equals(AdSize.FLUID);
+    }
+
+    @Override
+    public void requestLayout() {
+        super.requestLayout();
+
+        if (isFluid()) {
+            post(new MeasureAndLayoutRunnable());
+        }
+    }
+
+    @Override
+    protected void onLayout(boolean changed, int left, int top, int right, int bottom) {
+        super.onLayout(changed, left, top, right, bottom);
+
+        if (isFluid()) {
+            post(new MeasureAndLayoutRunnable());
+        }
+    }
+
+    private static void measureAndLayout(View view, int width, int height) {
+        int left = 0;
+        int top = 0;
+
+        view.measure(width, height);
+        view.layout(left, top, left + width, top + height);
+        view.requestLayout();
+        view.invalidate();
+        view.forceLayout();
+    }
+
+    int cachedWidth = 0;
+    int cachedHeight = 0;
+
+    private void updateLayout() {
+        try {
+            if (!isFluid()) {
+                return;
+            }
+
+            if (adView == null) {
+                return;
+            }
+
+            View parent = (View) adView.getParent();
+
+            if (parent == null) {
+                return;
+            }
+
+            int width = parent.getWidth();
+            int height = parent.getHeight();
+
+            if (cachedWidth == width && cachedHeight == height) {
+                return;
+            }
+
+            cachedWidth = width;
+            cachedHeight = height;
+
+            measureAndLayout(adView, width, height);
+
+            ViewGroup child = (ViewGroup) adView.getChildAt(0);
+
+            if (child != null) {
+                measureAndLayout(child, width, height);
+
+                ViewGroup webView = (ViewGroup) child.getChildAt(0);
+
+                if (webView != null) {
+                    measureAndLayout(webView, width, height);
+
+                    ViewGroup internalChild = (ViewGroup) webView.getChildAt(0);
+
+                    if (internalChild != null) {
+                        measureAndLayout(internalChild, width, height);
+
+                        ViewGroup leafNode = (ViewGroup) internalChild.getChildAt(0);
+
+                        if (leafNode != null) {
+                            measureAndLayout(leafNode, width, height);
+                        }
+                    }
+                }
+            }
+        } catch (Exception exception) {
```

Comment:
no logging?

Reply:
This is only safeguard, this is non-domain specific component. This should really never go there, but ust in case as the ads should never crash the host apps 😉